src/PersonObjects/Player/PlayerObjectGeneralMethods.tsx

```diff
@@ -1850,6 +1861,14 @@ export function quitJob(this: IPlayer, company: string, _sing = false): void {
   if (this.companyName === company) {
     this.companyName = this.hasJob() ? Object.keys(this.jobs)[0] : "";
   }
+
+  if (this.sleeves !== null) {
+    for (const sleeve of this.sleeves) {
+      if (sleeve.currentTask === SleeveTaskType.Company && sleeve.currentTaskLocation === company) {
+        sleeve.finishTask(this);
+      }
+    }
+  }
```

Comment:
it feels like this should be handled by the sleeves "quitJob" 

as wont this just quit your sleeves job when you quit as the player at company X?

Reply:
Yeah, I'm aware. The whole working system is a mess right now.
It does works; but it's spaghetti coding.

I'm considering discarding theses changes for the time being, and wait for the working system rework (It's already on hydro's TO-DO).

Reply:
Reverted that change.